### ISSUE TITLE
Issue in Iranian mobile number  

### ISSUE THREAD

**Issue by mshubhamtyagi** [tanansatpal/ngx-mat-intl-tel-input#46]
Hi Team, 

When I choose Iran country code +98. After choosing the country code when I enter digit of mobile number. It reset automatically after 5 digit but complete number have 9 digit.

**Comment by otejiri** [comment]
below code is how I solved the issue. Go to **node_modules** -> **ngx-mat-intl-tel-input** -> **__ivy_ngcc__** -> **fesm2015** and edit the **ngx-mat-intl-tel-input.js** file. Just replace where you have const phoneNumberValidator

```
const phoneNumberValidator = (control) => {
    const error = { validatePhoneNumber: true };
    let numberInstance;
    if (control.value) {
        try {
            numberInstance = parsePhoneNumber(control.value);

        }
        catch (e) {
            control.setValue(null);
            return error;
        }
        if (numberInstance.nationalNumber.length > 10) {
         return error;
        }
        if (numberInstance && !numberInstance.isValid()) {
            if (numberInstance.nationalNumber.length > 10) {
                return error;
            }
            if (!control.touched) {
                control.markAsTouched();
            }
            return error;
        }
    }
};
```

**Comment by mshubhamtyagi** [comment]
@otejiri I don't think this is a good solution. If someone removes the node_modules then we have to change it again. If we change the code on the local machine when we live the website on the server we have to change it on the server.

**Comment by tanansatpal** [comment]
Fixed in #49 . Will be merging and publishing it shortly
